File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxi = 0
        n = len(height)
        left = 0
        right = n-1
        while left < right:
            area = min(height[left],height[right]) * (right-left)
            if area > maxi:
                maxi = area
            if height[left] > height[right]:
                right -= 1
            else:
                left += 1
        return maxi

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int low = 0, high = height.size() - 1;
        int maxArea = 0;
        
        while(low < high){
            int length = min(height[low], height[high]);
            int width = high - low;
            int curLeng = length * width;
            maxArea = max(maxArea, curLeng);
            
            if(height[low] < height[high]){
                low++;
            }else{
                high--;
            }
        }
        return maxArea;
        
    }
};